design/one-pager-change-logs.md

```diff
@@ -0,0 +1,410 @@
+volume between the main provider and sidecar containers, and enables the alpha
+feature with the `--enable-change-logs` flag.
+
+```yaml
+apiVersion: pkg.crossplane.io/v1beta1
+kind: DeploymentRuntimeConfig
+metadata:
+  name: enable-change-logs
+spec:
+  deploymentTemplate:
+    spec:
+      selector: {}
+      template:
+        spec:
+          containers:
+          - name: package-runtime
+            args:
+            - --enable-change-logs
+            volumeMounts:
+            - name: change-log-vol
+              mountPath: /var/run/change-logs
+          - name: change-log-sidecar
+            image: crossplane/change-log-sidecar:latest
+            volumeMounts:
+            - name: change-log-vol
+              mountPath: /var/run/change-logs
+          volumes:
+          - name: change-log-vol
+            emptyDir: {}
+```
+
+### Implementation Rollout
+
+The general sequence of implementation work to roll this feature out is
+described below.  As previously mentioned, the plan does not include any updates
+to core Crossplane to support this feature, as we will rely on a
+`DeploymentRuntimeConfig` to perform the injection of the sidecar container and
+the setup of a shared volume between the main provider and sidecar containers.
+The key code changes we will make are in `crossplane-runtime`, a new
+`change-logs-sidecar` repo, the Upjet framework and template, and the ecosystem
+of providers.
+
+* `crossplane-runtime`:
+  * Protobuf types are defined for change log entries and a gRPC client/server
+    to send these entries from client to server
+  * Managed reconciler is updated to generate change log entries, send them to
+  the gRPC server (i.e. sidecar container), and to include additional calls to
+  `Observe()` after each CUD operation to capture the "after" state of the entry
+* `change-logs-sidecar`
+  * A new repository is created to build and publish the image used in the sidecar container
+  * Consumes the protobuf types from `crossplane-runtime`
+  * Starts the gRPC server and listens for change log entries
+  * Writes all entries to `stdout` (i.e. pod logs)
+* [Upjet framework](https://github.com/crossplane/upjet):
+  * Controller templates are updated to use the new `crossplane-runtime` managed
+    reconciler logic and initialize it with a gRPC client
+* [Upjet provider
+  template](https://github.com/crossplane/upjet-provider-template):
+  * Consumes this new Upjet framework version
+  * [`main.go`](https://github.com/crossplane/upjet-provider-template/blob/main/cmd/provider/main.go)
+    is updated to create and initialize the gRPC client and then pass it into
+    the setup for all of its controllers
+* The major upjet based providers (e.g. `provider-upjet-aws`,
+  `provider-upjet-gcp`, `provider-upjet-azure`, etc.)
+  * Updated with changes similar to the upjet provider template (consumes new
+    Upjet version and creates gRPC client to pass into its controllers)
```

Comment:
A PR against at least one native / non upjet provider could be helpful. This could be used by the community as an example to add support to their provider.

Reply:
Agreed! Actually for development simplicity, I have been iterating on https://github.com/crossplane-contrib/provider-kubernetes/compare/main...jbw976:provider-kubernetes:change-logs.  So that will definitely be a PR that I open 😉 

I'll udpate this design doc to make that known too.